File: main/src/main/java/lavaplayer/source/youtube/YoutubeMixLoader.java
package lavaplayer.source.youtube;

import lavaplayer.tools.io.HttpInterface;
import lavaplayer.track.AudioTrack;
import lavaplayer.track.AudioTrackInfo;
import lavaplayer.track.AudioTrackCollection;

import java.util.function.Function;

public interface YoutubeMixLoader {
    AudioTrackCollection load(
        HttpInterface httpInterface,
        String mixId,
        String selectedVideoId,
        Function<AudioTrackInfo, AudioTrack> trackFactory
    );
}

File: main/src/main/java/lavaplayer/source/youtube/music/YoutubeMixLoader.kt
package lavaplayer.source.youtube.music

import lavaplayer.tools.io.HttpInterface
import lavaplayer.track.AudioTrackCollection
import lavaplayer.track.AudioTrackFactory

interface YoutubeMixLoader {
    fun load(
        httpInterface: HttpInterface,
        mixId: String,
        selectedVideoId: String?,
        trackFactory: AudioTrackFactory
    ): AudioTrackCollection
}
